src/events.rs

```diff
@@ -162,43 +177,72 @@ pub struct NewViewEvent {
 }
 
-/// The replica received a [proposal](crate::messages::Proposal) for the replica's current view from the
+/// The replica broadcasted an [advance view](crate::pacemaker::messages::AdvanceView) message to all
+/// peers.
+pub struct AdvanceViewEvent {
+    pub timestamp: SystemTime,
+    pub advance_view: AdvanceView,
+}
+
+/// The replica broadcasted a [timeout vote](crate::pacemaker::messages::TimeoutVote) message to all
+/// peers.
+pub struct TimeoutVoteEvent {
+    pub timestamp: SystemTime,
+    pub timeout_vote: TimeoutVote,
+}
+
+/// The replica received a [proposal](crate::hotstuff::messages::Proposal) for the replica's current view from the
 /// leader of the view, identifiable by its [public key](ed25519_dalek::VerifyingKey).
 pub struct ReceiveProposalEvent {
     pub timestamp: SystemTime,
     pub origin: VerifyingKey,
     pub proposal: Proposal,
 }
 
-/// The replica received a [nudge](crate::messages::Nudge) for the replica's current view from the
+/// The replica received a [nudge](crate::hotstuff::messages::Nudge) for the replica's current view from the
 /// leader of the view, identifiable by its [public key](ed25519_dalek::VerifyingKey).
 pub struct ReceiveNudgeEvent {
     pub timestamp: SystemTime,
     pub origin: VerifyingKey,
     pub nudge: Nudge,
 }
 
-/// The replica received a [vote](crate::messages::Vote) for the replica's current view from another
+/// The replica received a [vote](crate::hotstuff::messages::Vote) for the replica's current view from another
 /// replica identifiable by its [public key](ed25519_dalek::VerifyingKey).
 pub struct ReceiveVoteEvent {
     pub timestamp: SystemTime,
     pub origin: VerifyingKey,
     pub vote: Vote,
 }
 
-/// The replica received a [new view](crate::messages::NewView) message for the current view from
+/// The replica received a [new view](crate::hotstuff::messages::NewView) message for the current view from
 /// another replica identifiable by its [public key](ed25519_dalek::VerifyingKey).
 pub struct ReceiveNewViewEvent {
     pub timestamp: SystemTime,
     pub origin: VerifyingKey,
     pub new_view: NewView,
 }
 
+/// The replica received an [adavance view](crate::pacemaker::messages::AdvanceView) message from
+/// another replica identifiable by its [public key](ed25519_dalek::VerifyingKey).
+pub struct ReceiveAdvanceViewEvent {
+    pub timestamp: SystemTime,
+    pub origin: VerifyingKey,
+    pub advance_view: AdvanceView,
+}
+
+/// The replica received a [timeout vote](crate::pacemaker::messages::TimeoutVote) for the replica's
+/// current view from another replica identifiable by its [public key](ed25519_dalek::VerifyingKey).
+pub struct ReceiveTimeoutVoteEvent {
+    pub timestamp: SystemTime,
+    pub origin: VerifyingKey,
+    pub timeout_vote: TimeoutVote,
+}
+
 /// The replica started a new view with a given [view number](crate::types::basic::ViewNumber) and
 /// a given leader identifiable by its [public key](ed25519_dalek::VerifyingKey).
```

Comment:
The `leader` field in this struct has been removed, but the itemdoc still suggests that the struct includes it.